### PR TITLE
#802 - DetailsCard - New component dedicated to details section

### PR DESCRIPTION
**Pull request by marcinsawicki**
Resolves: #802 

## Description
New `DetailsCard` component, and `DetailsCardInfo` sub-component.

## Storybook

https://feature-802--613a8e945a5665003a05113b.chromatic.com/?path=/docs/components-detailscard--docs

## Checklist

**Obligatory:**

- [x] Self review (use this as your final check for proposed changes before requesting the review)
- [x] Add reviewers (`livechat/design-system`)
- [x] Add correct label
- [x] Assign pull request with the correct issue

### CODE REVIEW

**Review by vladko-uxds — APPROVED** [review]
👍

File: packages/react-components/src/components/DetailsCard/DetailsCard.module.scss [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  &__float-button {
+    position: absolute;
+    top: var(--spacing-1);
+    right: var(--spacing-1);
+    opacity: 0;
+    z-index: 1;
+  }
+
+  &__content-wrapper {
+    transition: all 0.2s ease;
```
Comment (VadymBezpalko) [inline packages/react-components/src/components/DetailsCard/DetailsCard.module.scss:61]: I've spotted a strange animation for hidden label section:
https://monosnap.com/file/L6EKItWHu63AEWGKcI8vdFFzTsVCGq
Maybe we can improve it somehow?

File: packages/react-components/src/components/DetailsCard/DetailsCard.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import { DetailsCardInfo } from './DetailsCardInfo';
+
+import './DetailsCard.stories.css';
+
+export default {
+  title: 'Components/DetailsCard',
+  component: DetailsCard,
+};
+
+export const Default = (): React.ReactElement => {
```
Comment (VadymBezpalko) [inline packages/react-components/src/components/DetailsCard/DetailsCard.stories.tsx:21]: I think we can divide it into separate stories, so we can be consistent with already existing examples.

File: packages/react-components/src/components/DetailsCard/DetailsCard.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  );
+
+  return (
+    <div className={mergedClassNames}>
+      <button
+        className={cx(
+          styles[`${baseClass}__button`],
+          hideLabelOnOpen && isOpen && styles[`${baseClass}__button--hide`]
+        )}
+        onClick={() => setIsOpen(!isOpen)}
```
Comment (VadymBezpalko) [inline packages/react-components/src/components/DetailsCard/DetailsCard.tsx:67]: While operating on the existing state we should use an updated function, so please create a separate handler `handleHideButtonClick`, something like this:
```
const handleHideButtonClick = () => setIsOpen(prevValue => !prevValue);
```

File: packages/react-components/src/components/DetailsCard/DetailsCard.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    withDivider && styles[`${baseClass}--with-divider`],
+    className
+  );
+
+  return (
+    <div className={mergedClassNames}>
+      <button
+        className={cx(
+          styles[`${baseClass}__button`],
+          hideLabelOnOpen && isOpen && styles[`${baseClass}__button--hide`]
```
Comment (VadymBezpalko) [inline packages/react-components/src/components/DetailsCard/DetailsCard.tsx:65]: I see that there are a few usages of this condition (`hideLabelOnOpen && isOpen`) - we can put it into a separate const

File: packages/react-components/src/components/DetailsCard/DetailsCard.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        </Heading>
+        <Icon
+          className={cx(
+            styles[`${baseClass}__button__icon`],
+            isOpen && styles[`${baseClass}__button__icon--open`]
+          )}
+          source={ChevronRight}
+        />
+      </button>
+      {hideLabelOnOpen && isOpen && (
```
Comment (VadymBezpalko) [inline packages/react-components/src/components/DetailsCard/DetailsCard.tsx:83]: separate const for `hideLabelOnOpen && isOpen`

File: packages/react-components/src/components/DetailsCard/DetailsCard.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        <Button
+          kind="float"
+          icon={<Icon source={ChevronDown} />}
+          className={styles[`${baseClass}__float-button`]}
+          onClick={() => setIsOpen(!isOpen)}
```
Comment (VadymBezpalko) [inline packages/react-components/src/components/DetailsCard/DetailsCard.tsx:88]: please use a handler from above comment here

**Review by VadymBezpalko — APPROVED** [review]

**Pull request merged**